services/src/main/java/org/exoplatform/addons/gamification/rest/RealizationsRest.java

```diff
@@ -104,17 +99,49 @@ public Response getAllRealizations(@Parameter(description = "result fromDate", r
     if (offset < 0) {
       return Response.status(Response.Status.BAD_REQUEST).entity("Offset must be 0 or positive").build();
     }
-    if (limit <= 0) {
-      return Response.status(Response.Status.BAD_REQUEST).entity("Limit must be positive").build();
+    if (returnType.isEmpty()) {
+      return Response.status(Response.Status.BAD_REQUEST).entity("returnType must not be empty").build();
     }
+    if (returnType.equals("json")) {
+      if (limit <= 0) {
+        return Response.status(Response.Status.BAD_REQUEST).entity("Limit must be positive").build();
+      }
 
-    try {
-      List<GamificationActionsHistoryDTO> gActionsHistoryList =
-          realizationsService.getAllRealizationsByFilter(filter, offset, limit);
-      return Response.ok(GamificationActionsHistoryMapper.toRestEntities(gActionsHistoryList)).build();
-    } catch (Exception e) {
-      LOG.warn("Error retrieving list of Realizations", e);
-      return Response.serverError().entity(e.getMessage()).build();
+      try {
+        List<GamificationActionsHistoryDTO> gActionsHistoryList = realizationsService.getRealizationsByFilter(filter,
```

Comment:
please refactor the REST code to reuse the same block outside the `if...else` and just make sure to make a special treatment when the returnType equals to `xls`. When it's (**not empty** && **not equal to `json` or `xls`**) the return 400 Bad Request with message: "_Unsupported returnType, possible values: xls or json_".